examples/player/simid_player.js

```diff
@@ -609,14 +624,27 @@ class SimidPlayer {
     console.log('The creative has asked for the player to ping ' + requestedUrlArray);
   }
 
+  /**
+   * Called when creative requests a change in duration of ad.
+   * @private
+   */
   onRequestChangeAdDuration(incomingMessage) {
-    if (this.requestedDuration_ != NO_REQUESTED_DURATION) {
-      // TODO: Support multiple change duration requests.
-      this.simidProtocol.reject(incomingMessage);
+    this.requestedDuration_  = incomingMessage.args['duration'];
+    //If requested duration is negative seconds, play entire ad
```

Comment:
There is actually a negative number that can be passed in to requested duration.

https://interactiveadvertisingbureau.github.io/SIMID/#simid-creative-requestChangeAdDuration

One way to do this is to allow -2 and reject anything else. The other would be to stop the ad with any other negative values.

Reply:
okay great, I altered the code to check for the -2 and rejected other values